product_docs/docs/pgd/5/manual/04-installing-software.mdx

```diff
@@ -0,0 +1,352 @@
+---
+title: Step 4 - Installing the PGD software
+navTitle: Installing PGD software
+deepToC: true
+---
+
+## Installing the PGD software
+
+With the repositories configured, you can now install the Postgres Distributed software. 
+These steps must be carried out on each host before proceeding to the next step.
+
+* **Install the packages**
+  * Install the PGD packages which include a server specific BDR package and generic PGD proxy and cli packages. (`edb-bdr5-<postgresversion>`, `edb-pgd5-proxy`, and `edb-pgd5-cli`)
+
+
+* **Ensure the Postgres database server has been initialized and started.**
+  * Use `systemctl status ` to check the service is running
+  * If not, initialise the database and start the service
+
+
+* **Configure the BDR extension**
+  * Add the BDR extension (`$libdir/bdr`) at the start of the shared_preload_libraries setting in `postgresql.conf`.
+  * Set the `wal_level` GUC variable to `logical` in `postgresql.conf`.
+  * Turn on commit timestamp tracking by setting `track_commit_timestamp` to `'on'` in `postgresql.conf`.
+  * Raise the maximum worker processes to 16 by setting `max_worker_processes` to `'16'` in `postgresql.conf`.
```

Comment:
bdr.get_min_required_worker_processes() only works when the system is up and it can see the topology.

It is also not only undocumented but a surprise to development and it is already out of date and underestimates required workers for OTEL and other elements.

How to effectively calculate the value, before the system has been configured, is under discussion and will likely result in new docs in the postgres settings (or even a calculator).

Reply:
See BDR-2404 for the reasoning around the function. According to senior project developers:
> That calculation can’t be documented simply, but needs to be defined as a formula.

Reply:
It already is defined as a formula in the Postgres configuration docs page which is referenced in this page. Also the value given here is calculated according to that formula for a three node single database installation. 

And again, the sql function is absolutely no use to a user who has no running system because it is only used in tpaexec test to verify a node has sufficient worker processes.

Reply:
Maybe we could add "The value of 16 is fine for a standard three-node cluster; the exact number of processes required is a complex calculation depending on topology and configuration which will be documented at a later time."
As you can see in BDR-2404 the original plan was to use that function inside the PGD CLI, as part of the sanity checks. When this will happen, we could add that "You can use the PGD CLI to find out the optimal number of processes on each node."

Reply:
It still doesn't give you a number to use in a configuration *before* you have run the configured cluster for the first time. There is now a note attached which directs the user to where the calculation is actually documented.